File: benchmark/c/cpw/even_fibonacci_numbers_sum/even_fibonacci_numbers_sum.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int even_fibonacci_numbers_sum ( int limit ) ;
int even_fibonacci_numbers_sum ( int limit ) {
  if ( limit < 2 ) return 0;
  long int ef1 = 0, ef2 = 2;
  long int sum = ef1 + ef2;
  while ( ef2 <= limit ) {
    long int ef3 = 4 * ef2 + ef1;
    if ( ef3 > limit ) break;
    ef1 = ef2;
    ef2 = ef3;
    sum += ef2;
  }
  return sum;
}

File: translation_result/else_bms/success/c/cpw/even_fibonacci_numbers_sum/gpt4/CAPR/even_fibonacci_numbers_sum.rs

fn even_fibonacci_numbers_sum(limit: i32) -> i32 {
    if limit < 2 {
        return 0;
    }
    let mut ef1: i64 = 0;
    let mut ef2: i64 = 2;
    let mut sum: i64 = ef1 + ef2;

    while ef2 <= limit as i64 {
        let ef3 = 4 * ef2 + ef1;
        if ef3 > limit as i64 {
            break;
        }
        ef1 = ef2;
        ef2 = ef3;
        sum += ef2;
    }
    
    sum as i32
}
